dds_web/api/project.py

```diff
@@ -977,3 +977,49 @@ def get(self):
 
         return_info = {"project_info": project_info}
         return return_info
+
+    @auth.login_required(role=["Unit Admin", "Unit Personnel", "Project Owner"])
+    @logging_bind_request
+    @json_required
+    @handle_validation_errors
+    def put(self):
+        """Update Project information."""
+        # Get project ID, project and verify access
+        project_id = dds_web.utils.get_required_item(obj=flask.request.args, req="project")
+        project = dds_web.utils.collect_project(project_id=project_id)
+        dds_web.utils.verify_project_access(project=project)
+
+        # get the now info items
+        json_input = flask.request.json
+        new_title = json_input.get("title")
+        new_description = json_input.get("description")
+        new_pi = json_input.get("pi")
+
+        # valide the PI email address
+        pi_validator = marshmallow.validate.Email(error="bla")
+        try:
+            pi_validator(new_pi)
+        except marshmallow.ValidationError as err:
+            message = str(err)
+            raise DDSArgumentError(message="The PI email is invalid")
+
+        # current date for date_updated
+        curr_date = dds_web.utils.current_time()
+
+        # update the items
+        if new_title:
+            project.title = new_title
+        if new_description:
+            project.description = new_description
```

Comment:
Validate these too

Reply:
just pushed a commit for title and description validation, not so sure if that's what you had in mind

Reply:
Yes this is what I had in mind! 👍🏻  Possible that we can define these validators and just use them in the project creation and here but that's another task, I can try to have a look at that tomorrow or Wednesday.